pages/signUp.js

```diff
@@ -0,0 +1,97 @@
+/**
+ *
+ * @author Dhanushka
+ * @version 1.0
+ * @since 02-26-2019
+ *
+ */
+var SignUp = function () {
+  var common = require('../commonScripts/common');
+
+  var signuptb = element(by.xpath('//a[@data-gtm-id="account-management-sign-up"]'));
+  var loginTb = element(by.xpath('//a[@data-gtm-id="account-management-log-in"]'));
+  var email = element(by.xpath('//input[@id="email"]'));
+  var confEmail = element(by.xpath('//input[@id="confirmEmail"]'));
+  var choosePwd = element(by.xpath('//input[@id="choosePassword"]'));
+  var confPwd = element(by.xpath('//input[@id="confirmPassword"]'));
+  var showHidePwd_choose = element(by.xpath('//input[@id="choosePassword"]/..//a[@class="btn pwReveal"]'));
+  var showHidePwd_conf = element(by.xpath('//input[@id="confirmPassword"]/..//a[@class="btn pwReveal"]'));
+  var checkTerms = element(by.xpath('//input[@id="terms"]'));
+  var checkTermsLabel = element(by.xpath('//label[@for="terms"]'));
+  var checkPrivacy = element(by.xpath('//input[@id="privacy"]'));
+  var checkPrivacyLabel = element(by.xpath('//label[@for="privacy"]'));
+  var checkMarketing = element(by.xpath('//input[@id="marketing"]'));
+  var checkMarketingLabel = element(by.xpath('//label[@for="marketing"]'));
+  var signUpBtn = element(by.xpath('//button[@data-gtm-id="account-management-sign-up-submit"]'));
+  var successlabel = element(by.xpath('//h2[@class="account-created__title"]'));
+
+
+  //this is to navigate signup page
+  this.get = function () {
+    browser.get('https://us.megabus.com/account-management/login');
+  };
+
+  //This is to verify select Sign up tab and validate the fields default status
+  this.SelectSignUpTab = function () {
+    signuptb.click();
+    console.log('Sign-Up tab is selected');
+  };
+
+  //This is to verify fields/labels/check boxes
+  this.VerifySignUpData = function (email1, confEmailWrng, confEmail1, choosePwd1, confPwdWrng1, confPwd1) {
```

Comment:
This can be written in test case since you will be using signup only once and others will not reuse this function